src/calculateRentalCost.js

```diff
@@ -4,7 +4,23 @@
  * @return {number}
  */
 function calculateRentalCost(days) {
-  // write code here
+  const DAILY_RENTAL_RATE = 40;
+  const LONG_TERM_DAYS = 7;
+  const LONG_TERM_DISCOUNT = 50;
+  const MEDIUM_TERM_DAYS = 3;
+  const MEDIUM_TERM_DISCOUNT = 20;
+
+  const baseCost = days * DAILY_RENTAL_RATE;
+
+  if (days >= LONG_TERM_DAYS) {
+    return baseCost - LONG_TERM_DISCOUNT;
+  }
+
+  if (days >= MEDIUM_TERM_DAYS) {
+    return baseCost - MEDIUM_TERM_DISCOUNT;
```

Comment:
Logical error: The medium-term discount is applied even if the long-term discount condition is met. This is because the function returns immediately after applying the medium-term discount. To fix this, you should reorder the conditions or ensure that only the largest applicable discount is applied.